tests/integration/targets/firewall_device/tasks/main.yaml

```diff
@@ -0,0 +1,60 @@
+- name: firewall_device
+  block:
+    - name: Create a Linode Instance
+      linode.cloud.instance:
+        api_token: '{{ api_token }}'
+        label: 'ansible-test-{{ ansible_date_time.epoch }}'
+        region: us-southeast
+        type: g6-standard-1
+        image: linode/alpine3.13
+        state: present
+      register: inst
+
+    - name: Create a Linode Firewall
+      linode.cloud.firewall:
+        api_token: '{{ api_token }}'
+        api_version: v4beta
+        label: 'ansible-test-{{ ansible_date_time.epoch }}'
+        devices: []
+        rules:
+          inbound: []
+          inbound_policy: DROP
+          outbound: []
+          outbound_policy: DROP
+        state: present
+      register: fw
+
+    - name: Add Device to Linode Firewall
+      linode.cloud.firewall_device:
+        api_token: '{{ api_token }}'
+        api_version: v4beta
+        firewall_id: '{{ fw.firewall.id }}'
+        entity_id: '{{ inst.instance.id }}'
+        entity_type: 'linode'
+        state: present
+      register: fw_device
+
+    - name: Assert firewall device added
+      assert:
+        that:
+          - fw_device.changed
```

Comment:
We might want to duplicate this test and assert `not fw_device.changed` to make sure we properly retrieve existing devices.